csrc/multidevice/executor.cpp

```diff
@@ -50,7 +50,7 @@ MultiDeviceExecutor::MultiDeviceExecutor(
     Communicator& comm,
     MultiDeviceExecutorParams params)
     : comm_(comm), params_(params) {
-  propagateShardings(fusion.get());
+  propagateShardingsAndSetAllocationDomain(fusion.get());
```

Comment:
Can we put a comment here saying that `propagateShardingsAndSetAllocationDomain needs to happen before `PreSegmenter` optimization passes? I think it's useful to have that as a note when we later decided to merge the code path.

I'm not sure if that applies to `insertReshardings` and `insertShardedAxisReordering` as well.

Reply:
So far applies to all of the (re)sharding passes!